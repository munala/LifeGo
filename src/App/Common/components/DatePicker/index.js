import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    alignSelf: 'stretch',
    flexDirection: 'column',
    alignItems: 'stretch',
    marginTop: 5,
  },
  textField: {
    display: 'flex',
    justifyContent: 'center',
    alignSelf: 'center',
    width: 'calc(100% - 40px)',
  },
});

const DatePicker = ({
  classes,
  label,
  defaultValue,
  onChange,
}) => (
  <form className={classes.container} noValidate>
    <TextField
      id="date"
      label={label}
      type="date"
      onChange={onChange}
      defaultValue={defaultValue}
      className={classes.textField}
      InputLabelProps={{
        shrink: true,
      }}
    />
  </form>
);

DatePicker.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  label: PropTypes.string.isRequired,
  defaultValue: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default withStyles(styles)(DatePicker);
