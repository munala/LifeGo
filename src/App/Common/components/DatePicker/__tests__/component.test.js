import React from 'react';
import { shallow } from 'enzyme';
import DatePicker from '../';

describe('DatePicker tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<DatePicker
      label="source"
      defaultValue=""
      onChange={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
