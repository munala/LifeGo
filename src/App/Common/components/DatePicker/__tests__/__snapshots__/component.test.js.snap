// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker tests renders without crashing 1`] = `
<DatePicker
  classes={
    Object {
      "container": "DatePicker-container-1",
      "textField": "DatePicker-textField-2",
    }
  }
  defaultValue=""
  label="source"
  onChange={[Function]}
/>
`;
